src/cli/batch.rs

```diff
@@ -159,30 +165,22 @@ async fn handle_batch_send_command(
 	}
 
 	// Submit batch transaction
-	let tx_hash = batch_transfer(&quantus_client, &keypair, transfers, tip_amount).await?;
+	let tx_hash =
+		batch_transfer(&quantus_client, &keypair, transfers, tip_amount, finalized).await?;
 
 	log_print!(
 		"✅ {} Batch transaction submitted! Hash: {:?}",
 		"SUCCESS".bright_green().bold(),
 		tx_hash
 	);
 
-	let success =
-		crate::cli::progress_spinner::wait_for_tx_confirmation(quantus_client.client(), tx_hash)
-			.await?;
-
-	if success {
-		log_info!("✅ Batch transaction confirmed and finalized on chain");
-		log_success!("🎉 {} Batch transaction confirmed!", "FINISHED".bright_green().bold());
+	log_success!("🎉 {} Batch transaction confirmed!", "FINISHED".bright_green().bold());
 
-		// Show updated balance
-		let new_balance = crate::cli::send::get_balance(&quantus_client, &from_account_id).await?;
-		let formatted_new_balance =
-			crate::cli::send::format_balance_with_symbol(&quantus_client, new_balance).await?;
-		log_print!("💰 New balance: {}", formatted_new_balance.bright_yellow());
-	} else {
```

Comment:
Let's keep error logging please

Reply:
error handling is moved to `submit_transaction` now (it's a bit generic, yeah). and also, previously this error would never show up because `wait_for_tx_confirmation` did not have a failure state (it just sleeped and returned success)